pydicom/valuerep.py

```diff
@@ -896,12 +919,10 @@ def __init__(
             self.original_string = val
             self._components = None
         else:
-            # handle None `val` as empty string
-            val = val or ''
-
-            # this is the decoded string - save the original string if
-            # available for easier writing back
-            self.original_string = original_string
+            # val: str
+            # `val` is the decoded person name value
+            # `original_string`  should be the original encoded value
+            self.original_string = cast(bytes, original_string)
             components = val.split('=')
```

Comment:
@mrbean-bremen one question I have is [6.2.1.2](http://dicom.nema.org/medical/dicom/current/output/chtml/part05/sect_6.2.html#sect_6.2.1.2) says:

> Delimiter characters "^" and "=" are taken from the character set specified by value 1 of the Attribute Specific Character Set (0008,0005). If Attribute Specific Character Set (0008,0005), value 1 is not present, the Default Character Repertoire ISO-IR 6 shall be used.

So does that mean the encoded components should be split using `'='.encode(encodings[0])` and similarly for the group delimiter?

Reply:
I don't think we've ever had a problem with these delimiters, but perhaps it can't hurt to use the encoded value?  Maybe the encoding is not changed in any character sets we've hit so far?

Wait, though, here it looks like splitting original bytes - maybe the correct way is to decode the entire string to Python `str` before this split, in which case it is internal unicode and the "=" (also Python `str`, unicode) would be the correct way.  Commenting here without really thinking it through properly ...

Reply:
Yeah this was just the example I found to show what I meant. It'd apply to the encoded `bytes`

Reply:
This was just an example to show what I was referring to, should've made that a bit clearer

Reply:
Sorry, missed that... I have to admit that I was completely unaware of this. I always assumed that ASCII is used for these delimiters. For most practical purposes, this is probably true, because ASCII is a subset of many encodings. But you are right, to do it correctly, we have to encode the delimiters.

Reply:
My thinking was that if it always devolves to the ASCII ``'='`` and ``'^'`` chars then there'd be no need to state that they should be taken from the first character set.